examples/cifar10/train.cpp

```diff
@@ -104,21 +116,75 @@ void train_cifar10(string data_dir_path, double learning_rate, ostream &log) {
                           n_train_epochs, on_enumerate_minibatch,
                           on_enumerate_epoch);
 
-  cout << "end training." << endl;
+  std::cout << "end training." << std::endl;
 
   // test and show results
-  nn.test(test_images, test_labels).print_detail(cout);
-
+  nn.test(test_images, test_labels).print_detail(std::cout);
   // save networks
-  ofstream ofs("cifar-weights");
+  std::ofstream ofs("cifar-weights");
   ofs << nn;
 }
 
+static core::backend_t parse_backend_name(const char *name) {
+  static const char *names[] = {
```

Comment:
this last looks better. However, we should provide simple examples with simple code and use "modern" interfaces as much as possible in order to let new users be confident with the library as fast as possible. That was my point.

@Randl up to you to update the code after seeing this discussion. Ping once you are done.